src/ray/core_worker/reference_count_test.cc

```diff
@@ -621,6 +621,15 @@ TEST_F(ReferenceCountTest, TestGetLocalityData) {
   ASSERT_EQ(locality_data_obj1->nodes_containing_object,
             absl::flat_hash_set<NodeID>({node1}));
 
+  // When node2 is dead, reference table shoule remove it from obj1's locations.
+  // And then GetLocalityData should only return node1.
+  rc->AddObjectLocation(obj1, node2);
+  locality_data_obj1 = rc->GetLocalityData(obj1);
```

Comment:
Nit: Confirm that `node2` was added to `obj1`'s locations, as a baseline. I know that this is already tested above on line 617, but this dead node filtering test case should confirm that we're going from `{node1, node2}` -> `{node1}` and this redundancy makes the test case self-contained.
```suggestion
  locality_data_obj1 = rc->GetLocalityData(obj1);
  ASSERT_EQ(locality_data_obj1->nodes_containing_object,
            absl::flat_hash_set<NodeID>({node1, node2}));
```